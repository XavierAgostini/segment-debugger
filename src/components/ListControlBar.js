import React from 'react'
import { Button, SegmentedControl, SearchInput } from 'evergreen-ui'

const toolBarStyle = {
  'WebKitBoxAlign': 'center',
  alignItems: 'center',
  borderBottom: '1px solid #d5dee6',
  display: '-webkit-box',
  display: '-ms-flexbox',
  display: 'flex',
  height: '72px',
  padding: '0 16px',
}
const buttonWrapper = {
  width: '160px',
  display: 'flex',
  margin: '0px -1px 0px 0px'
}
const searchBarWrapper = {
  position: 'relative',
  minWidth: '40px',
  minHeight: '40px',
  display: 'flex',
  margin: '0px 16px'
}
export default class ListControlBar extends React.Component {
  state={
    options: [
      { label: 'Live', value: 'live' },
      { label: 'Pause', value: 'pause' }
    ],
    value: 'live'
  }
  handleButtonChange = (value) => {``
    this.setState({ value })
    const isPaused = this.state.value === 'live'
    this.props.handlePauseChange(isPaused)
  }
  handleSearchChange = (e) => {
    const val = e.target.value
    this.props.handleSearchFilter(val)
  }

  render() {
    return (
      <div style={toolBarStyle}>
        <SegmentedControl
          width={160}
          height={40}
          options={this.state.options}
          value={this.state.value}
          onChange={value => this.handleButtonChange(value)}
        />
        <SearchInput
          name="searchBox"
          marginX='16px'
          placeholder='Type to search...'
          width='360px'
          height={40}
          onChange={this.handleSearchChange}
        />
        <Button
        alignItems='center'
        children='Advanced'
        display='inline-flex'
        disabled={true}
        is='button'
        justifyContent='center'
        secondary='true'
        >
          Advanced
        </Button>
      </div>    
    )
  }
}