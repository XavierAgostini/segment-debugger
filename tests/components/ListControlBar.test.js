import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ListControlBar from '../../src/components/ListControlBar'

let wrapper, handlePauseChange, handleSearchFilter

beforeEach(() => {
  handlePauseChange = jest.fn()
  handleSearchFilter = jest.fn()
  wrapper = shallow(<ListControlBar handlePauseChange={handlePauseChange} handleSearchFilter={handleSearchFilter}/>)
})

describe('<ListControlBar/>', () => {
  test('should render ListControlBar as expected', () => {
    expect(wrapper).toMatchSnapshot()
  })
})