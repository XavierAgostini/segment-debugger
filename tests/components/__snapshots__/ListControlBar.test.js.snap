// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListControlBar/> should render ListControlBar as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListControlBar
    handlePauseChange={[MockFunction]}
    handleSearchFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SegmentedControl
          height={40}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Live",
                "value": "live",
              },
              Object {
                "label": "Pause",
                "value": "pause",
              },
            ]
          }
          value="live"
          width={160}
        />,
        <_class
          height={40}
          marginX="16px"
          name="searchBox"
          onChange={[Function]}
          placeholder="Type to search..."
          width="360px"
        />,
        <_class
          alignItems="center"
          disabled={true}
          display="inline-flex"
          is="button"
          justifyContent="center"
          secondary="true"
        >
          Advanced
        </_class>,
      ],
      "style": Object {
        "WebKitBoxAlign": "center",
        "alignItems": "center",
        "borderBottom": "1px solid #d5dee6",
        "display": "flex",
        "height": "72px",
        "padding": "0 16px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "height": 40,
          "onChange": [Function],
          "options": Array [
            Object {
              "label": "Live",
              "value": "live",
            },
            Object {
              "label": "Pause",
              "value": "pause",
            },
          ],
          "value": "live",
          "width": 160,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "height": 40,
          "marginX": "16px",
          "name": "searchBox",
          "onChange": [Function],
          "placeholder": "Type to search...",
          "width": "360px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alignItems": "center",
          "children": "Advanced",
          "disabled": true,
          "display": "inline-flex",
          "is": "button",
          "justifyContent": "center",
          "secondary": "true",
        },
        "ref": null,
        "rendered": "Advanced",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SegmentedControl
            height={40}
            onChange={[Function]}
            options={
              Array [
                Object {
                  "label": "Live",
                  "value": "live",
                },
                Object {
                  "label": "Pause",
                  "value": "pause",
                },
              ]
            }
            value="live"
            width={160}
          />,
          <_class
            height={40}
            marginX="16px"
            name="searchBox"
            onChange={[Function]}
            placeholder="Type to search..."
            width="360px"
          />,
          <_class
            alignItems="center"
            disabled={true}
            display="inline-flex"
            is="button"
            justifyContent="center"
            secondary="true"
          >
            Advanced
          </_class>,
        ],
        "style": Object {
          "WebKitBoxAlign": "center",
          "alignItems": "center",
          "borderBottom": "1px solid #d5dee6",
          "display": "flex",
          "height": "72px",
          "padding": "0 16px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "height": 40,
            "onChange": [Function],
            "options": Array [
              Object {
                "label": "Live",
                "value": "live",
              },
              Object {
                "label": "Pause",
                "value": "pause",
              },
            ],
            "value": "live",
            "width": 160,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "height": 40,
            "marginX": "16px",
            "name": "searchBox",
            "onChange": [Function],
            "placeholder": "Type to search...",
            "width": "360px",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alignItems": "center",
            "children": "Advanced",
            "disabled": true,
            "display": "inline-flex",
            "is": "button",
            "justifyContent": "center",
            "secondary": "true",
          },
          "ref": null,
          "rendered": "Advanced",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
